wcfsetup/install/files/lib/system/box/BoxHandler.class.php

```diff
@@ -185,6 +189,56 @@ public function addBoxToPageAssignments($boxIdentifier, array $pageIdentifiers,
                 $statement->execute([$box->boxID, $page->pageID, $visible ? 1 : 0]);
             }
         }
+
+        $conditionObjectTypeID = ObjectTypeCache::getInstance()->getObjectTypeIDByName(
+            'com.woltlab.wcf.condition.box',
+            'com.woltlab.wcf.page'
+        );
+        $oldCondition = [];
+        foreach ($box->getVisibilityConditions() as $condition) {
+            if ($condition->objectTypeID === $conditionObjectTypeID) {
+                $oldCondition[] = $condition;
+                break;
+            }
+        }
+
```

Comment:
Add a safety check that you indeed found a matching condition.

Reply:
Why? It is totally fine, if the array is empty.